Controller/AjaxController.php

```diff
@@ -301,4 +304,57 @@ protected function getTokensAction(Request $request)
 
         return $this->sendJsonResponse($dataArray);
     }
+
+    /**
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     *
+     * @throws \Exception
+     */
+    public function pendingEventsAction()
+    {
+        $objectId = $this->request->request->get('objectId') ? (int) ($this->request->request->get('objectId')) : null;
+        $data     = [];
+        $total    = 0;
+        $eventIds = [];
+
+        if (!empty($objectId)) {
+            $em      = $this->dispatcher->getContainer()->get('doctrine.orm.default_entity_manager');
+            /** @var ClientEventHelper $clientEventHelper */
+            $clientEventHelper = $this->get('mautic.contactclient.helper.client_event');
+            if ($events = $clientEventHelper->getAllClientEvents($objectId)) {
+                $eventIds = array_keys($events);
+            }
+
+            /** @var ContactClientRepository $repo */
+            $clientRepo    = $em->getRepository(\MauticPlugin\MauticContactClientBundle\Entity\ContactClient::class);
+
+            $events       = $clientRepo->getPendingEventsData($objectId, $eventIds);
+            foreach ($events as $event) {
+                $total = $total + (int) $event['count'];
+            }
+
+            $headers    = [
+                '',
+                'mautic.contactclient.pending.events.campaign',
+                '',
+                'mautic.contactclient.pending.events.event',
+                'mautic.contactclient.pending.events.count',
+            ];
+
+            foreach ($headers as $header) {
+                $data['columns'][] = [
+                    'title' => $this->translator->trans($header),
+                ];
+            }
+            $values = [];
+            foreach ($events as $event) {
+                $values[] = array_values($event);
+            }
+        }
+        $data['events'] = $values;
```

Comment:
`$values` is defined only inside the if statement, but it's used even if the `if` statement didn't happen.

As long as it's intended that `$values` is null when `empty($objectId)`, that's ok. But PHP will throw notices.

Reply:
no, that should be at the top with the rest of the declarations